openesdh-repo/src/main/java/dk/openesdh/repo/services/documents/CaseDocumentCopyServiceImpl.java

```diff
@@ -220,6 +228,89 @@ public NodeRef copyDocFolderItemRetainVersionLabels(CaseFolderItem item, NodeRef
+            NodeRef sourceCommentsFolder = getCommentsFolder(sourceNode);
+            nodeService.getChildAssocs(sourceCommentsFolder, ContentModel.ASSOC_CONTAINS, null).stream()
+                    .map(ChildAssociationRef::getChildRef)
+                    .forEach(commentRef -> moveComment(commentRef, targetCommentsFolder));
+            return null;
+        });
+    }
+
+    @Override
+    public void detachCaseDocument(NodeRef documentRef, NodeRef newOwner, String comment) {
+        NodeRef mainDocNodeRef = documentService.getMainDocument(documentRef);
+        tr.runInTransaction(() -> {
+            authorityFilesService.move(mainDocNodeRef, newOwner, comment);
+            nodeService.setType(mainDocNodeRef, ContentModel.TYPE_CONTENT);
+            moveDocumentComments(documentRef, mainDocNodeRef);
+            nodeService.deleteNode(documentRef);
+            return null;
+        });
+    }
+
+    @Override
+    public void copyCaseDocumentsFromTempAttachments(NodeRef caseNodeRef) {
+        if (!nodeService.hasAspect(caseNodeRef, OpenESDHModel.ASPECT_OE_TEMP_ATTACHMENTS)) {
+            return;
+        }
+
+        String jsonTempAttachmens = (String) nodeService.getProperty(caseNodeRef,
+                OpenESDHModel.PROP_OE_TEMP_ATTACHMENTS);
+
+        if (StringUtils.isEmpty(jsonTempAttachmens)) {
+            nodeService.removeAspect(caseNodeRef, OpenESDHModel.ASPECT_OE_TEMP_ATTACHMENTS);
+            return;
+        }
+
+        NodeRef caseDocsFolder = caseService.getDocumentsFolder(caseNodeRef);
+        try {
+            JSONArray tempAttachments = new JSONArray(new JSONTokener(jsonTempAttachmens));
+            for (int i = 0; i < tempAttachments.length(); i++) {
+                copyCaseDocFromTempAttachment(tempAttachments.getJSONObject(i), caseDocsFolder);
+            }
+        } catch (JSONException e) {
+            throw new AlfrescoRuntimeException("Error parsing temp attachments json", e);
+        }
+
+        nodeService.removeAspect(caseNodeRef, OpenESDHModel.ASPECT_OE_TEMP_ATTACHMENTS);
+    }
+
+    private NodeRef copyCaseDocFromTempAttachment(JSONObject tempAttachment, NodeRef targetFolderRef)
+            throws JSONException {
+        String tmpFileName = tempAttachment.getString(OeFilesService.TMP_FILE_NAME);
+        String fileName = tempAttachment.getString(OeFilesService.FILE_NAME);
+        String mimeType = tempAttachment.getString(OeFilesService.MIME_TYPE);
+        NodeRef docType = new NodeRef(tempAttachment.getString(WebScriptParams.DOC_TYPE));
+        NodeRef docCategory = new NodeRef(tempAttachment.getString(WebScriptParams.DOC_CATEGORY));
+
+        File[] files = TempFileProvider.getTempDir().listFiles((file, name) -> name.equals(tmpFileName));
+        if (files.length == 0) {
+            throw new AlfrescoRuntimeException("Temp attachment file not found " + tmpFileName + " " + fileName);
+        }
+
+        Map<QName, Serializable> props = new HashMap<>();
```

Comment:
all the following could be replaced to call to documentService.createDocumentFile